### ISSUE TITLE
Synthesize wav

### ISSUE THREAD

**Issue by zuoxiang95** [keithito/tacotron#6]
hi, i use this project generate some good result in my logdir.
Some name looks like step-100000-audio.wav. I wonder if it was synthesize in 100000 steps.Thanks~~

**Comment by keithito** [comment]
Yes, `step-100000-audio.wav` is audio synthesized at global step 100000 of training. Note that this isn't a good evaluation of the model at step 100000 because (a) training uses teacher forcing so the RNN gets fed the previous ground truth output as input when predicting the current output, and (b) it's running on training data rather than something the model hasn't seen before.

To evaluate the model, you can use `eval.py` or `demo_server.py`, passing the checkpoint that you want to use. For example: `python eval.py --checkpoint ~/tacotron/logs-tacotron/model.ckpt-100000`.

**Comment by zuoxiang95** [comment]
Thanks a lot , i have another question.  When the test input txt is very long ,  the synthesize wav will be noise in the end.Do you know why this happened?

**Comment by keithito** [comment]
It depends. Especially early on in the training process, the model may not have learned to correctly align the input and output yet. Or it could be that it hasn't learned to generate silence after reaching the end. Both of these tend to improve if you keep training longer.

**Comment by zuoxiang95** [comment]
But the synthesized wav can read the first half of the text clear.

**Comment by zuoxiang95** [comment]
for example, my input txt is : 
Scientists at the CERN laboratory say they have discovered a new particle.Generative adversarial network or variational auto-encoder.
The synthesized wav can read "Scientists at the CERN laboratory say they have discovered a new particle.Generative " clear, but the end of sentence can not be hear clear.

**Comment by keithito** [comment]
Can you attach or link to the synthesized wav?

**Comment by zuoxiang95** [comment]
here is [link](https://qdcache03.baidupcs.com/file/53a5a73594f9efca3ec740b6aa05d4f4?bkt=p3-140053a5a73594f9efca3ec740b6aa05d4f4316fdd3300000003cec8&xcode=785f7e3b353aaff505779f1092946640357627d5edcf3e880b2977702d3e6764&fid=1235506260-250528-769128675576471&time=1500864583&sign=FDTAXGERLBHS-DCb740ccc5511e5e8fedcff06b081203-d3mpEAsphuQtv0tslO8RR3HRRZ0%3D&to=p5&size=249544&sta_dx=249544&sta_cs=15&sta_ft=wav&sta_ct=0&sta_mt=0&fm2=MH,Guangzhou,Netizen-anywhere,,guangxi,cnc&newver=1&newfm=1&secfm=1&flow_ver=3&pkey=140053a5a73594f9efca3ec740b6aa05d4f4316fdd3300000003cec8&sl=76480590&expires=8h&rt=sh&r=265207328&mlogid=4741800420915092770&vuk=1235506260&vbdid=3042649595&fin=eval-123000-0.wav&fn=eval-123000-0.wav&rtype=1&iv=0&dp-logid=4741800420915092770&dp-callid=0.1.1&hps=1&csl=80&csign=pAOicYjcjnXJabq5C67sW7snC%2Fc%3D&so=0&ut=6&uter=4&serv=0&by=themis)

**Comment by zuoxiang95** [comment]
[wav.zip](https://github.com/keithito/tacotron/files/1168681/wav.zip)

**Comment by zuoxiang95** [comment]
here is my synthesized wav.
[wav.zip](https://github.com/keithito/tacotron/files/1168683/wav.zip)

**Comment by keithito** [comment]
Thanks! Try setting the max_iters hyperaparameter to a larger value at eval time, for example by passing the flag `--hparams="max_iters=300"`. This will allow the decoder to run for more iterations.

Even then, a model trained on the LJ Speech Dataset might not generalize to extremely long sequences since the longest wav in the dataset is 10 seconds. It's probably best to try to chop up your inputs into shorter sequences. You could also try building a training set with longer sequences by concatenating multiple texts (and their corresponding wavs) and seeing if that would work.

**Comment by zuoxiang95** [comment]
Thank you very much. This is a good idea, i will try it in the next few days.

**Comment by sonach** [comment]
step-xxxx-audio.wav is not a good evalution for the training. In my training, the step-9000-audio.wav is OK, but use mode.cpkt-9000 to synthesize text produce very bad wav.

**Comment by zuoxiang95** [comment]
In the code , training uses teacher forcing. So step-xxxx-audio.wav can not be used to evaluate the training.
